src/utils/DateUtils.ts

```diff
@@ -346,7 +346,7 @@ export class DateUtils {
     }
 
     if (options.predefinedFormat) {
-      return moment(date).format(options.predefinedFormat);
+      return moment(date).format(options.predefinedFormat.replace(/yyyy/g, 'YYYY'));
```

Comment:
I think it would be helpful to use a separate function for transforming from lowercase to uppercase `Y`. It would provide context and reduce the repetition between this line and line 279. e.g.

```
function getMomentJsFormat(format: string) {
    // yyyy was used to format dates before implementing moment.js, which only recognizes YYYY.
    return format.replace(/yyyy/g, 'YYYY');
}
```